apps/mull-api/src/app/user/user.resolver.ts

```diff
@@ -37,7 +47,7 @@ export class UserResolver {
 
   @Query(/* istanbul ignore next */ () => [Friend])
   async friends(@AuthenticatedUser() id: number) {
-    const friends = await this.userService.getFriends(id);
+    const friends = this.filterTrueFriends(id, await this.userService.getFriends(id));
```

Comment:
`friends` on purpose takes only the friends you have in your friend list and doesn't check for 2 way btw. You should change your call in the frontend from `friends` to `getTrueFriends`